src/common/config.cc

```diff
@@ -473,6 +473,31 @@ void md_config_t::parse_env(unsigned entity_type,
       _set_val(values, tracker, dir, *o, CONF_ENV, &err);
     }
   }
+
+  // Apply pod memory limits:
+  //
+  // There are two types of resource requests: `limits` and `requests`.
+  //
+  // - Requests: Used by the K8s scheduler to determine on which nodes to
+  //   schedule the pods. This helps spread the pods to different nodes. This
+  //   value should be conservative in order to make sure all the pods are
+  //   schedulable.
+  // - Limits: At runtime, K8s will use the limits to see if the pod is using
+  //   too many resources. In that case, the pod will be killed/restarted
+  //   automatically if the pod goes over the limit. This should be higher than
+  //   what is specified for requests (potentially much higher). 
+  //
+  // Here are the documented best practices: https://kubernetes.io/docs/tasks/configure-pod-container/assign-cpu-resource/#motivation-for-cpu-requests-and-limits
+  //
+  // When the operator creates the CephCluster CR, it will need to generate the
+  // desired requests and limits. As long as we are conservative in our choice
+  // for requests and generous with the limits we should be in a good place to
+  // get started.
+  //
+  // The support in Rook is already there for applying the limits as seen in these links.
+  //
+  // Rook docs on the resource requests and limits: https://rook.io/docs/rook/v1.0/ceph-cluster-crd.html#cluster-wide-resources-configuration-settings
+  // Example CR settings: https://github.com/rook/rook/blob/6d2ef936698593036185aabcb00d1d74f9c7bfc1/cluster/examples/kubernetes/ceph/cluster.yaml#L90
   if (auto pod_req = getenv("POD_MEMORY_REQUEST"); pod_req) {
```

Comment:
This POD_MEMORY_REQUEST should only correspond to the "limits", so would it be better to focus this comment around that? Feel free to modify my comment however needed. :)

Reply:
> This POD_MEMORY_REQUEST should only correspond to the "limits"

I must be confused, I thought it was the opposite!

Reply:
Yeah, the naming matches up with the wrong option. What K8s calls the `limit`, we're requesting the pod not to exceed or else it might be terminated.

Reply:
Your description is still confusing.  

IIUC,

- *REQUEST* is the size we expect the pod to be, and what we schedule against.  (and what we should tell the OSD to target.)
- *LIMIT* is the cgroup limit that will trigger the kernel's OOM killer

Reply:
Yes, that's clear, thanks. I'm sure i was just confused about the intent of `POD_MEMORY_REQUEST`. Does this variable match what we expect the REQUEST to be? I was thinking the ceph daemon would care more about the LIMIT.

Reply:
Oh I see.  In this case, we care about the request.  The process will try its best to limit itself to ``osd_memory_target`` but it may overshoot it slightly, or oscillate around that value.  So we don't want to have a hard limit close to the target.  And we want to target what k8s is scheduling against.